<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Visualization of the differentially expressed genes - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Visualization of the differentially expressed genes";
    var mkdocs_page_input_path = "DEseq2visu.md";
    var mkdocs_page_url = "/springday/DEseq2visu/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Cutadapt/">Optional filtering of reads on sequence quality</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class="">
                    
    <a class="" href="../star/">RNA STAR</a>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../intro_counting/">Introduction to read counting</a>
                </li>
                <li class="">
                    
    <a class="" href="../strandness/">Estimation of the strandness</a>
                </li>
                <li class="">
                    
    <a class="" href="../count/">Count the number of reads per annotated gene</a>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../volcano/">Volcano Plot</a>
                </li>
                <li class=" current">
                    
    <a class="current" href="./">Visualization of the differentially expressed genes</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#visualisation-of-differential-expressions">Visualisation of differential expressions</a></li>
    
        <ul>
        
            <li><a class="toctree-l4" href="#extract-the-most-differentially-expressed-genes">Extract the most differentially expressed genes</a></li>
        
            <li><a class="toctree-l4" href="#extract-the-normalized-counts-of-the-most-differentially-expressed-genes">Extract the normalized counts of the most differentially expressed genes</a></li>
        
        </ul>
    

    </ul>
                </li>
                <li class="">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../GO-tool/">GO</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Galaxy Workflows</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../workflow_intro/">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../workflow_use/">Example</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Analysis of the differential gene expression &raquo;</li>
        
      
    
    <li>Visualization of the differentially expressed genes</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/ARTbio/springday/edit/master/docs/DEseq2visu.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><img alt="" src="../images/galaxylogo.png" /></p>
<h1 id="visualisation-of-differential-expressions">Visualisation of differential expressions</h1>
<p>Now we would like to extract the most differentially expressed genes due to the treatment,
and then visualize them using an heatmap of the normalized counts and also
the z-score for each sample.</p>
<p>We will proceed in several steps:</p>
<ul>
<li>Extract the most differentially expressed genes using the DESeq2 summary file</li>
<li>Extract the normalized counts for these genes for each sample, using the normalized count file generated by DESeq2</li>
<li>Plot the heatmap of the normalized counts</li>
<li>Compute the Z score of the normalized counts</li>
<li>Plot the heatmap of the Z score of the genes</li>
</ul>
<h2 id="extract-the-most-differentially-expressed-genes">Extract the most differentially expressed genes</h2>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ol>
<li>Select the tool <code>Filter data on any column using simple expressions</code> to extract genes with a significant change in gene expression (adjusted p-value below 0.05) between treated and untreated samples:<ol>
<li><code>Filter</code>: the DESeq2 result file</li>
<li><code>With following condition</code>: c7&lt;0.05</li>
</ol>
</li>
</ol>
<p>The file with the independent filtered results can be used for further downstream analysis
as it excludes genes with only few read counts as these genes will not be considered as significantly differentially expressed.</p>
<p>The generated file contains too many genes (632/STAR, ) to get a meaningful heatmap. Therefore, in the next step,
we will take only the genes with an absolute fold change &gt; 2 (log2(fold change) &gt; 1)</p>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ol>
<li>Select the tool <code>Filter data on any column using simple expressions</code><ol>
<li><code>Filter</code>: the differentially expressed genes (output of previous <code>Filter</code> tool)</li>
<li><code>With following condition</code>: abs(c3)&gt;1</li>
</ol>
</li>
</ol>
<p>We now have a table with 84/STAR, /HISAT2 lines corresponding to the most differentially expressed genes.
And for each of the gene, we have its id, its mean normalized counts (averaged over all
samples from both conditions), its log2FC and other information.</p>
<p>We could plot the log2FC for the different genes, but here we would like to look at a
heatmap of expression for these genes in the different samples. So we need to extract the
normalized counts for these genes.</p>
<p>We will join the normalized count table generated by DESeq2 with the table we just generated,
to conserve only the lines corresponding to the most differentially expressed genes.</p>
<h2 id="extract-the-normalized-counts-of-the-most-differentially-expressed-genes">Extract the normalized counts of the most differentially expressed genes</h2>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ul>
<li>
<p>Create a Pasted Entry from the header line of the Filter output:</p>
<ol>
<li>Copy the header of the final Filter output</li>
<li>Using the Upload tool select Paste/Fetch data and paste the copied data</li>
<li><em>Set the Type to tabular</em> and select Start to upload a new Pasted Entry</li>
</ol>
</li>
</ul>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ul>
<li>Concatenate datasets tool to add this header line to the Filter output:<ol>
<li>select the <code>Concatenate datasets tail-to-head</code> tool</li>
<li>select the Pasted entry dataset</li>
<li><code>+ Insert Dataset</code></li>
<li>select the final <code>Filter output</code></li>
</ol>
</li>
</ul>
<p>This ensures that the table of most differentially expressed genes has a header line and can be used in the next step.</p>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ul>
<li>
<p>join the normalized count table generated by DESeq2 with the table we just generated,
to conserve only the lines corresponding to the most differentially expressed genes</p>
<ol>
<li>select the <code>Join two Datasets side by side on a specified field</code> tool<ul>
<li><code>Join</code>: the Normalized counts file (output of DESeq2 tool)</li>
<li><code>using column</code>: Column: 1</li>
<li><code>with</code>: most differentially expressed genes (output of the Concatenate tool tool)</li>
<li><code>and column</code>: Column: 1</li>
<li><code>Keep lines of first input that do not join with second input</code>: No</li>
<li><code>Keep the header lines</code>: Yes</li>
</ul>
</li>
</ol>
</li>
</ul>
<p>The generated file has more columns than we need for the heatmap. In addition to the columns
with mean normalized counts, there is the log2FC and other information.
We need to remove the extra columns.</p>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ul>
<li>
<p>Cut tool to extract the columns with the gene ids and normalized counts:</p>
<ol>
<li>Select the <code>Cut columns from a table</code>tool<ul>
<li><code>Cut columns</code>: c1-c8</li>
<li><code>Delimited by</code>: Tab</li>
<li><code>From</code>: the joined dataset (output of Join two Datasets tool)</li>
</ul>
</li>
</ol>
</li>
</ul>
<p>We now have a table with 85 lines (the most differentially expressed genes)
and the normalized counts for these genes in the 7 samples.</p>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ul>
<li>
<p>Plot the heatmap of the normalized counts of these genes for the samples</p>
<ol>
<li>Select the <code>heatmap2</code> tool to plot the heatmap:<ul>
<li><code>Input should have column headers</code>: the generated table (output of Cut tool)</li>
<li><code>Data transformation</code>:    <strong>Log2(value+1)</strong> transform my data</li>
<li><code>Enable data clustering</code>: Yes</li>
<li><code>Labeling columns and rows</code>: Label columns and not rows</li>
<li><code>Coloring groups</code>: Blue to white to red</li>
</ul>
</li>
</ol>
</li>
</ul>
<p>You should obtain something similar to:</p>
<p><img alt="" src="../images/cluster.png" /></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../GO-intro/" class="btn btn-neutral float-right" title="Introduction">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../volcano/" class="btn btn-neutral" title="Volcano Plot"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/ARTbio/springday/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../volcano/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../GO-intro/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
