<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>RNA STAR - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "RNA STAR";
    var mkdocs_page_input_path = "star.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Cutadapt/">Optional filtering of reads on sequence quality</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class=" current">
                    
    <a class="current" href="./">RNA STAR</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#rna-star-option-for-50-of-attendees">RNA STAR (option for 50 % of attendees)</a></li>
    
        <ul>
        
            <li><a class="toctree-l4" href="#rename-your-datasets">Rename your datasets !</a></li>
        
        </ul>
    

    </ul>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../intro_counting/">Introduction to read counting</a>
                </li>
                <li class="">
                    
    <a class="" href="../strandness/">Estimation of the strandness</a>
                </li>
                <li class="">
                    
    <a class="" href="../count/">Count the number of reads per annotated gene</a>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEseq2visu/">Visualization of the differentially expressed genes</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../GO-tool/">GO</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../conclusion/">Conclusion</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Mapping &raquo;</li>
        
      
    
    <li>RNA STAR</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><img alt="" src="../images/galaxylogo.png" /></p>
<h1 id="rna-star-option-for-50-of-attendees">RNA STAR (option for 50 % of attendees)</h1>
<p>For information to set proper value for STAR parameters:</p>
<p><img alt="" src="../images/multiqc_samples_length.png" /></p>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ol>
<li>create a new history and name it <code>RNA STAR</code></li>
<li>Import the 11 datasets from the RNAseq data library to this <code>RNA STAR</code> history, plus the Drosophila_melanogaster.BDGP6.95.gtf file</li>
<li>Select the <code>RNA STAR</code> tool with the following parameters to map your reads on the reference genome:<ol>
<li><code>Single-end or paired-end reads</code>: Single-end</li>
<li><code>RNA-Seq FASTQ/FASTA file</code> (multiple datasets button), Cmd-shift Select:<ul>
<li><code>GSM461176_untreat_single.fastq.gz</code></li>
<li><code>GSM461179_treat_single.fastq.gz</code></li>
</ul>
</li>
<li><code>Custom or built-in reference genome:</code> Use a built-in index</li>
<li><code>Reference genome with or without an annotation:</code> use genome reference without builtin gene-model</li>
<li><code>Select reference genome:</code> Drosophila Melanogaster (dm6)</li>
<li><code>Gene model (gff3,gtf) file for splice junctions:</code> the imported Drosophila_melanogaster.BDGP6.95.gtf</li>
<li><code>Length of the genomic sequence around annotated junctions:</code> 44 (This parameter should be length of reads - 1, see above table from fastQC/multiQC analysis)</li>
</ol>
</li>
<li><code>Execute</code></li>
</ol>
<hr />
<p><img alt="" src="../images/redo.png" /></p>
<p>Redo the STAR run with</p>
<p><strong>3.2</strong>    Select the <code>RNA STAR</code> tool with the following parameters to map your reads on the reference genome:
   <code>RNA-Seq FASTQ/FASTA file</code> (as multiple datasets), Cmd-shift Select:
          - <code>GSM461182_untreat_single.fastq.gz</code></p>
<p><strong>3.7</strong>    <code>Length of the genomic sequence around annotated junctions:</code> 74 (This parameter should be length of reads - 1, see above table from fastQC/multiQC analysis)</p>
<hr />
<p><img alt="" src="../images/redo.png" /></p>
<p>Redo a last STAR run for paired-end datasets</p>
<ol>
<li>With the following parameters to map your reads on the reference genome:<ol>
<li><code>Single-end or paired-end reads</code>: Paired-end (as multiple datasets)</li>
<li><code>RNA-Seq FASTQ/FASTA file, forward reads</code> (multiple datasets button), Cmd-shift Select:<ul>
<li><code>GSM461177_1_untreat_paired.fastq.gz</code></li>
<li><code>GSM461178_1_untreat_paired.fastq.gz</code></li>
<li><code>GSM461180_1_treat_paired.fastq.gz</code></li>
<li>`GSM461181_1_treat_paired.fastq.gz</li>
</ul>
</li>
<li><code>RNA-Seq FASTQ/FASTA file, forward reads</code> (multiple datasets button), Cmd-shift Select:<ul>
<li><code>GSM461177_2_untreat_paired.fastq.gz</code></li>
<li><code>GSM461178_2_untreat_paired.fastq.gz</code></li>
<li><code>GSM461180_2_treat_paired.fastq.gz</code></li>
<li>`GSM461181_2_treat_paired.fastq.gz</li>
</ul>
</li>
<li><code>Custom or built-in reference genome:</code> Use a built-in index</li>
<li><code>Reference genome with or without an annotation:</code> use genome reference without builtin gene-model</li>
<li><code>Select reference genome:</code> Drosophila Melanogaster (dm6)</li>
<li><code>Gene model (gff3,gtf) file for splice junctions:</code> the imported Drosophila_melanogaster.BDGP6.95.gtf</li>
<li><code>Length of the genomic sequence around annotated junctions:</code> <strong>36</strong> (This parameter should be length of reads - 1, see above table from fastQC/multiQC analysis)</li>
</ol>
</li>
<li><code>Execute</code></li>
</ol>
<h2 id="rename-your-datasets">Rename your datasets !</h2>
<hr />
<p><img alt="" src="../images/rename_datasets.png" /></p>
<p>You need now to rename you datasets to facilitate your downstream analysis.</p>
<p>Be quiet and focus ! No hurry, this is an <em>important task</em> in the analysis.</p>
<ol>
<li>
<p>Search and select datasets with RNA STAR <img alt="" src="../images/search_star.png" /></p>
</li>
<li>
<p>Click on the info icon <img alt="" src="../images/info.png" /> of both <code>log</code> and <code>bam</code> files</p>
</li>
<li>
<p>Copy the name or one of the two names of the datasets as shown bellow
<img alt="" src="../images/copy.png" /></p>
</li>
<li>Now click on the pencil icon of the <em><em>same</em></em> dataset
<img alt="" src="../images/pencil.png" /></li>
<li>Paste your text in the <code>Name</code> field of the dataset
<img alt="" src="../images/paste.png" /></li>
<li>Edit your text as follow for <code>log</code> files
<img alt="" src="../images/edit_log.png" /></li>
<li>
<p>Edit your text as follow for <code>bam</code> files
<img alt="" src="../images/edit_bam.png" /></p>
</li>
<li>
<p>repeat <em>ad lib</em> for all <code>log</code> and <code>bam</code> files </p>
</li>
</ol>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../hisat2/" class="btn btn-neutral float-right" title="HiSAT2">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../mapping/" class="btn btn-neutral" title="Mapping"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../mapping/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../hisat2/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
