<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>GO - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "GO";
    var mkdocs_page_input_path = "GO-tool.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Cutadapt/">Optional filtering of reads on sequence quality</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class="">
                    
    <a class="" href="../star/">RNA STAR</a>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../intro_counting/">Introduction to read counting</a>
                </li>
                <li class="">
                    
    <a class="" href="../strandness/">Estimation of the strandness</a>
                </li>
                <li class="">
                    
    <a class="" href="../count/">Count the number of reads per annotated gene</a>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEseq2visu/">Visualization of the differentially expressed genes</a>
                </li>
                <li class=" current">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3 current">
                    
    <a class="current" href="./">GO</a>
    <ul class="subnav">
            
    </ul>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <a class="" href="../conclusion/">Conclusion</a>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Analysis of functional enrichment among the differentially expressed genes &raquo;</li>
        
      
        
          <li>Analysis of the differential gene expression &raquo;</li>
        
      
    
    <li>GO</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p>Hands-on: Prepare the datasets for GOSeq
Compute tool with
“Add expression”: bool(c7&lt;0.05)
param-file “as a new column to”: the DESeq2 result file
Cut tool with
“Cut columns”: c1,c8
“Delimited by”: Tab
param-file “From”: the output of the Compute tool
Change Case tool with
param-file “From”: the output of the previous Cut tool
“Change case of columns”: c1
“Delimited by”: Tab
“To”: Upper case
We just generated the first input for goseq. We need as second input for goseq, the gene lengths. We can use there the gene length generated by featureCounts tool and reformat it a bit.</p>
<p>Change Case tool with
param-file “From”: the feature lengths (output of featureCounts tool)
“Change case of columns”: c1
“Delimited by”: Tab
“To”: Upper case
We have now the two required input files for goseq.</p>
<p>hands_on Hands-on: Perform GO analysis
goseq tool with
“Differentially expressed genes file”: first file generated by Change Case tool on previous step
“Gene lengths file”: second file generated by Change Case tool on previous step
“Gene categories”: Get categories
“Select a genome to use”: Fruit fly (dm6)
“Select Gene ID format”: Ensembl Gene ID
“Select one or more categories”: GO: Cellular Component, GO: Biological Process, GO: Molecular Function
goseq generates a big table with the following columns for each GO term:</p>
<p>category: GO category
over_rep_pval: p-value for over representation of the term in the differentially expressed genes
under_rep_pval: p-value for under representation of the term in the differentially expressed genes
numDEInCat: number of differentially expressed genes in this category
numInCat: number of genes in this category
term: detail of the term
ontology: MF (Molecular Function - molecular activities of gene products), CC (Cellular Component - where gene products are active), BP (Biological Process - pathways and larger processes made up of the activities of multiple gene products)
p.adjust.over_represented: p-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure
p.adjust.under_represented: p-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure
To identify categories significantly enriched/unenriched below some p-value cutoff, it is necessary to use the adjusted p-value.</p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../conclusion/" class="btn btn-neutral float-right" title="Conclusion">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../GO-intro/" class="btn btn-neutral" title="Introduction"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../GO-intro/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../conclusion/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
