<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Optional filtering of reads on sequence quality - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Optional filtering of reads on sequence quality";
    var mkdocs_page_input_path = "Cutadapt.md";
    var mkdocs_page_url = "/springday/Cutadapt/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class=" current">
                    
    <a class="current" href="./">Optional filtering of reads on sequence quality</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#filtering-datasets-to-remove-or-trim-low-quality-sequences">Filtering datasets to remove or trim low quality sequences</a></li>
    
        <ul>
        
            <li><a class="toctree-l4" href="#this-step-is-optional-and-should-be-performed-by-50-of-attendees">This step is optional and should be performed by 50% of attendees.</a></li>
        
            <li><a class="toctree-l4" href="#cutadapt-with-single-reads">Cutadapt with single reads</a></li>
        
            <li><a class="toctree-l4" href="#cutadapt-with-paired-end-reads">Cutadapt with paired-end reads</a></li>
        
            <li><a class="toctree-l4" href="#run-multiqc-on-cutadapt-jobs">Run MultiQC on Cutadapt jobs</a></li>
        
        </ul>
    

    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class="">
                    
    <a class="" href="../star/">RNA STAR</a>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../intro_counting/">Introduction to read counting</a>
                </li>
                <li class="">
                    
    <a class="" href="../strandness/">Estimation of the strandness</a>
                </li>
                <li class="">
                    
    <a class="" href="../count/">Count the number of reads per annotated gene</a>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../volcano/">Volcano Plot</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEseq2visu/">Visualization of the differentially expressed genes</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../GO-tool/">GO</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Galaxy Workflows</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../workflow_intro/">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../workflow_use/">Example</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Quality filtering &raquo;</li>
        
      
    
    <li>Optional filtering of reads on sequence quality</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/ARTbio/springday/edit/master/docs/Cutadapt.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><img alt="" src="../images/galaxylogo.png" /></p>
<h1 id="filtering-datasets-to-remove-or-trim-low-quality-sequences">Filtering datasets to remove or trim low quality sequences</h1>
<h2 id="this-step-is-optional-and-should-be-performed-by-50-of-attendees">This step is optional and should be performed by 50% of attendees.</h2>
<h2 id="cutadapt-with-single-reads">Cutadapt with single reads <img alt="" src="../images/tool_small.png" /></h2>
<hr />
<ol>
<li>Create a new history <code>Cutapdapt</code></li>
<li>Copy the fastq files from the RNAseq data library to this new history</li>
<li>Select the <code>Cutadapt</code> tool</li>
<li>Start with selecting <code>Single-end</code> in the <code>Single-end or Paired-end reads?</code> menu</li>
<li>Select the multiple datasets button for this menu</li>
<li>Cmd-Click for discontinuous multiple selection of <code>single</code> fastq.gz files (3 datasets)</li>
<li><code>Filter Options</code></li>
<li><code>Minimum length</code>: 20</li>
<li><code>Read Modification Options</code></li>
<li><code>Quality cutoff</code>: 20</li>
<li><code>Output Options</code></li>
<li><code>Report</code>: Yes</li>
<li>Do not change the other available parameters and click <code>Execute</code></li>
</ol>
<hr />
<h2 id="cutadapt-with-paired-end-reads">Cutadapt with paired-end reads <img alt="" src="../images/tool_small.png" /></h2>
<hr />
<p>Repeat the same procedure as above, except that you select <code>Paired-end</code>in step 4:
Re-Run the tool using the re-run button on one Cutadapt instance and just select <code>Paired-end</code>
instead of <code>Single-end</code></p>
<ul>
<li>
<p>Then you have two input boxes, one for file #1 and one for file #2.</p>
</li>
<li>
<p>In the box <code>file #1</code> click the <code>multiple datasets</code> button and carefully Select
the fastq.gz files with the <code>_1</code> suffix</p>
</li>
<li>
<p>In the box <code>file #2</code> click the <code>multiple datasets</code> button and carefully Select
the fastq.gz files with the <code>_2</code> suffix</p>
</li>
<li>
<p>Do not change the other parameters (they are set to the same value as previously because
you used the re-run button).</p>
</li>
<li>
<p>Click the <code>Execute</code> button</p>
</li>
</ul>
<hr />
<h2 id="run-multiqc-on-cutadapt-jobs">Run MultiQC on Cutadapt jobs <img alt="" src="../images/tool_small.png" /></h2>
<hr />
<ol>
<li>Select <code>MultiQC</code> tool</li>
<li>Select <code>Cutadapt/Trim Galore!</code> in the menu <code>Which tool was used generate logs?</code></li>
<li>Cmd-Select the <code>Report</code> datasets generated by Cutadapt</li>
<li>Press <code>Execute</code></li>
<li>
<p>Now, the borring but essential job: Rename carefully the <code>Output</code> datasets generated
by Cutadapt. To do so, help yourself with the <code>Info</code> button at the bottom of dataset green
boxes.</p>
<p>Example: Rename <code>Cutadapt on data 10 and data 9: Read 2 Output</code> in <code>GSM461181_2_treat_paired.fastq.gz</code></p>
</li>
<li>
<p>Trash the 11 unfiltered/trimmed fastq.gz files. This is important to avoid mixing
filtered and non filtered datasets in the next steps.</p>
</li>
</ol>
<hr />
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../mapping/" class="btn btn-neutral float-right" title="Mapping">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../outline_conclusion/" class="btn btn-neutral" title="Experimental procedures affect downstream analyses"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/ARTbio/springday/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../outline_conclusion/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../mapping/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
