<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Introduction to read counting - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Introduction to read counting";
    var mkdocs_page_input_path = "intro_counting.md";
    var mkdocs_page_url = "/springday/intro_counting/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Cutadapt/">Optional filtering of reads on sequence quality</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class="">
                    
    <a class="" href="../star/">RNA STAR</a>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class=" current">
                    
    <a class="current" href="./">Introduction to read counting</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#counting-strategy">Counting strategy</a></li>
    
        <ul>
        
            <li><a class="toctree-l4" href="#count-the-number-of-reads-per-annotated-gene">Count the number of reads per annotated gene</a></li>
        
            <li><a class="toctree-l4" href="#counting-tools">Counting tools</a></li>
        
        </ul>
    

    </ul>
                </li>
                <li class="">
                    
    <a class="" href="../strandness/">Estimation of the strandness</a>
                </li>
                <li class="">
                    
    <a class="" href="../count/">Count the number of reads per annotated gene</a>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../volcano/">Volcano Plot</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEseq2visu/">Visualization of the differentially expressed genes</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../GO-tool/">GO</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Galaxy Workflows</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../workflow_intro/">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../workflow_use/">Example</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Analysis of the differential gene expression &raquo;</li>
        
      
    
    <li>Introduction to read counting</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/ARTbio/springday/edit/master/docs/intro_counting.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <h1 id="counting-strategy">Counting strategy <img alt="" src="../images/lamp.png" /></h1>
<p><img alt="" src="../images/readcounts.png" /></p>
<h2 id="count-the-number-of-reads-per-annotated-gene">Count the number of reads per annotated gene</h2>
<p>To compare the expression of single genes between different conditions
(e.g. with or without <em>Pasilla</em> depletion), an essential first step is to quantify
the number of reads per gene.</p>
<p><img alt="" src="../images/gene_counting.png" /></p>
<p>From the image above, we can compute:</p>
<h3 id="number-of-reads-per-exons">Number of reads per exons</h3>
<table>
<thead>
<tr>
<th>Gene</th>
<th>Exon</th>
<th>Number of reads</th>
</tr>
</thead>
<tbody>
<tr>
<td>gene1</td>
<td>exon1</td>
<td>3</td>
</tr>
<tr>
<td>gene1</td>
<td>exon2</td>
<td>2</td>
</tr>
<tr>
<td>gene2</td>
<td>exon1</td>
<td>3</td>
</tr>
<tr>
<td>gene2</td>
<td>exon2</td>
<td>4</td>
</tr>
<tr>
<td>gene2</td>
<td>exon3</td>
<td>3</td>
</tr>
</tbody>
</table>
<ul>
<li>The gene1 has 4 reads, not 5 (gene1 - exon1 + gene1 - exon2) because of the splicing of the last read.</li>
<li>The gene2 has 6 reads (3 spliced reads)</li>
</ul>
<h2 id="counting-tools">Counting tools</h2>
<p>Two main tools could be used for that: HTSeq-count
(<a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">Anders et al, Bioinformatics, 2015</a>)
or featureCounts (<a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">Liao et al, Bioinformatics, 2014</a>).
FeatureCounts is considerably faster and requires far less computational resources,
so we will use it here.</p>
<p>In principle, the counting of reads overlapping with genomic features is a fairly
simple task. But there are some details that need to be given to featureCounts:
for example the strandness.</p>
<p><img alt="" src="../images/sequencing_strategies.png" /></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../strandness/" class="btn btn-neutral float-right" title="Estimation of the strandness">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../visu_map/" class="btn btn-neutral" title="Visualisation of read mapping"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/ARTbio/springday/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../visu_map/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../strandness/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
