<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Estimation of the strandness - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Estimation of the strandness";
    var mkdocs_page_input_path = "strandness.md";
    var mkdocs_page_url = "/springday/strandness/";
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Cutadapt/">Optional filtering of reads on sequence quality</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class="">
                    
    <a class="" href="../star/">RNA STAR</a>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../intro_counting/">Introduction to read counting</a>
                </li>
                <li class=" current">
                    
    <a class="current" href="./">Estimation of the strandness</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#estimation-of-the-strandness">Estimation of the strandness</a></li>
    
        <ul>
        
            <li><a class="toctree-l4" href="#use-of-infer-experiment-tool">Use of Infer Experiment tool</a></li>
        
        </ul>
    

    </ul>
                </li>
                <li class="">
                    
    <a class="" href="../count/">Count the number of reads per annotated gene</a>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../volcano/">Volcano Plot</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEseq2visu/">Visualization of the differentially expressed genes</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../GO-tool/">GO</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Galaxy Workflows</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../workflow_intro/">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../workflow_use/">Example</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Analysis of the differential gene expression &raquo;</li>
        
      
    
    <li>Estimation of the strandness</li>
    <li class="wy-breadcrumbs-aside">
      
        <a href="https://github.com/ARTbio/springday/edit/master/docs/strandness.md"
          class="icon icon-github"> Edit on GitHub</a>
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><img alt="" src="../images/galaxylogo.png" /></p>
<h1 id="estimation-of-the-strandness">Estimation of the strandness</h1>
<p>In practice, with Illumina RNA-seq protocols you will most likely deal with either:</p>
<ul>
<li>
<p>Unstranded RNAseq data</p>
</li>
<li>
<p>Stranded RNA-seq data produced with - kits and dUTP tagging (ISR)</p>
</li>
</ul>
<p>This information should be provided with your FASTQ files, ask your sequencing facility!</p>
<p>If not, try to find it on the site where you downloaded the data or
in the corresponding publication.</p>
<p>Another option is to estimate these parameters with a tool called <code>Infer Experiment</code> from
the <code>RSeQC</code> tool suite. This tool takes the output of your mappings (BAM files), selects
a subsample of your reads and compares their genome coordinates and strands with those of
the reference gene model (from an annotation file).</p>
<p>Based on the strand of the genes, it can gauge whether sequencing is strand-specific,
and if so, how reads are stranded.</p>
<h2 id="use-of-infer-experiment-tool">Use of <code>Infer Experiment</code> tool</h2>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<h3 id="convert-gtf-to-bed12-tool-to-convert-the-gtf-file-to-bed"><code>Convert GTF to BED12</code> tool to convert the GTF file to BED</h3>
<ol>
<li>Go to your history <code>STAR</code> or <code>HISAT2</code></li>
<li>Select the tool <code>Convert GTF to BED12</code><ol>
<li><code>GTF File to convert</code>: Drosophila_melanogaster.BDGP6.95.gtf</li>
</ol>
</li>
<li><code>Execute</code></li>
</ol>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<h3 id="infer-experiment-tool-to-determine-the-library-strandness"><code>Infer Experiment</code> tool to determine the library strandness</h3>
<ol>
<li>In the same history <code>STAR</code> or <code>HISAT2</code></li>
<li>Select the tool <code>Infer Experiment</code><ol>
<li><code>Input .bam file</code>: mapped.bam files (outputs of RNA STAR or HISAT2 tools)</li>
<li><code>Reference gene model</code> : BED12 file (output of Convert GTF to BED12 tool)</li>
<li><code>Number of reads sampled from SAM/BAM file (default = 200000)</code>: 200000</li>
</ol>
</li>
<li><code>Execute</code></li>
</ol>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<h3 id="summarize-results-with-multiqc-tool">Summarize results with <code>MultiQC</code> tool</h3>
<ol>
<li>Select the tool <code>MultiQC</code></li>
<li>Which tool was used generate logs?<ul>
<li><code>RSeQC</code></li>
</ul>
</li>
<li>RSeQC output (Type of RSeQC output?)<ul>
<li>infer_experiment</li>
</ul>
</li>
<li>Select the 7 datasets of type <code>Infer Experiment on ...</code></li>
<li><code>Execute</code></li>
</ol>
<hr />
<p><img alt="" src="../images/lamp.png" /></p>
<p>Infer Experiment tool generates one file with information on:</p>
<ul>
<li>Paired-end or single-end library</li>
<li>Fraction of reads failed to determine</li>
<li>2 lines:<ul>
<li>For single-end<ul>
<li>Fraction of reads explained by “++,–” (SF in previous figure)</li>
<li>Fraction of reads explained by “+-,-+” (SR in previous figure)</li>
</ul>
</li>
<li>For paired-end<ul>
<li>Fraction of reads explained by “1++,1–,2+-,2-+” (SF in previous figure)</li>
<li>Fraction of reads explained by “1+-,1-+,2++,2–” (SR in previous figure)</li>
</ul>
</li>
</ul>
</li>
</ul>
<p>If the two “Fraction of reads explained by” numbers are close to each other (<em>i.e.</em> a mix of SF and SR),
we conclude that the library is not a strand-specific dataset (U in previous figure).</p>
<p>As it is sometimes quite difficult to find out which settings correspond to those of
other programs, the following table might be helpful to identify the library type:</p>
<table>
<thead>
<tr>
<th>Library type</th>
<th>Infer Experiment</th>
<th>TopHat</th>
<th>HISAT</th>
<th>htseq-count</th>
<th>featureCounts</th>
</tr>
</thead>
<tbody>
<tr>
<td>Paired-End (PE) - SF</td>
<td>1++,1–,2+-,2-+</td>
<td>FR Second Strand</td>
<td>Second Strand F/FR</td>
<td>yes</td>
<td>Forward (1)</td>
</tr>
<tr>
<td>PE - SR</td>
<td>1+-,1-+,2++,2–</td>
<td>FR First Strand</td>
<td>First Strand R/RF</td>
<td>reverse</td>
<td>Reverse (2)</td>
</tr>
<tr>
<td>Single-End (SE) - SF</td>
<td>+,–</td>
<td>FR Second Strand</td>
<td>Second Strand F/FR</td>
<td>yes</td>
<td>Forward (1)</td>
</tr>
<tr>
<td>SE - SR</td>
<td>+-,-+</td>
<td>FR First Strand</td>
<td>First Strand R/RF</td>
<td>reverse</td>
<td>Reverse (2)</td>
</tr>
<tr>
<td>PE, SE - U</td>
<td>undecided</td>
<td>FR Unstranded</td>
<td>default</td>
<td>no</td>
<td>Unstranded (0)</td>
</tr>
</tbody>
</table>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../count/" class="btn btn-neutral float-right" title="Count the number of reads per annotated gene">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../intro_counting/" class="btn btn-neutral" title="Introduction to read counting"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
          <a href="https://github.com/ARTbio/springday/" class="fa fa-github" style="float: left; color: #fcfcfc"> GitHub</a>
      
      
        <span><a href="../intro_counting/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../count/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
