

Hands-on: Prepare the datasets for GOSeq
Compute tool with
“Add expression”: bool(c7<0.05)
param-file “as a new column to”: the DESeq2 result file
Cut tool with
“Cut columns”: c1,c8
“Delimited by”: Tab
param-file “From”: the output of the Compute tool
Change Case tool with
param-file “From”: the output of the previous Cut tool
“Change case of columns”: c1
“Delimited by”: Tab
“To”: Upper case
We just generated the first input for goseq. We need as second input for goseq, the gene lengths. We can use there the gene length generated by featureCounts tool and reformat it a bit.

Change Case tool with
param-file “From”: the feature lengths (output of featureCounts tool)
“Change case of columns”: c1
“Delimited by”: Tab
“To”: Upper case
We have now the two required input files for goseq.

hands_on Hands-on: Perform GO analysis
goseq tool with
“Differentially expressed genes file”: first file generated by Change Case tool on previous step
“Gene lengths file”: second file generated by Change Case tool on previous step
“Gene categories”: Get categories
“Select a genome to use”: Fruit fly (dm6)
“Select Gene ID format”: Ensembl Gene ID
“Select one or more categories”: GO: Cellular Component, GO: Biological Process, GO: Molecular Function
goseq generates a big table with the following columns for each GO term:

category: GO category
over_rep_pval: p-value for over representation of the term in the differentially expressed genes
under_rep_pval: p-value for under representation of the term in the differentially expressed genes
numDEInCat: number of differentially expressed genes in this category
numInCat: number of genes in this category
term: detail of the term
ontology: MF (Molecular Function - molecular activities of gene products), CC (Cellular Component - where gene products are active), BP (Biological Process - pathways and larger processes made up of the activities of multiple gene products)
p.adjust.over_represented: p-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure
p.adjust.under_represented: p-value for over representation of the term in the differentially expressed genes, adjusted for multiple testing with the Benjamini-Hochberg procedure
To identify categories significantly enriched/unenriched below some p-value cutoff, it is necessary to use the adjusted p-value.

