<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  
  <link rel="shortcut icon" href="../img/favicon.ico">
  <title>Count the number of reads per annotated gene - Galaxy Spring day 2019</title>
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700|Roboto+Slab:400,700|Inconsolata:400,700' rel='stylesheet' type='text/css'>

  <link rel="stylesheet" href="../css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../css/theme_extra.css" type="text/css" />
  <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css">
  
  <script>
    // Current page data
    var mkdocs_page_name = "Count the number of reads per annotated gene";
    var mkdocs_page_input_path = "count.md";
    var mkdocs_page_url = null;
  </script>
  
  <script src="../js/jquery-2.1.1.min.js" defer></script>
  <script src="../js/modernizr-2.8.3.min.js" defer></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
  <script>hljs.initHighlightingOnLoad();</script> 
  
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
      <div class="wy-side-nav-search">
        <a href=".." class="icon icon-home"> Galaxy Spring day 2019</a>
        <div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
	<ul class="current">
	  
          
            <li class="toctree-l1">
		
    <a class="" href="..">Home</a>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Pretreatments</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../uploads/">Data upload</a>
                </li>
                <li class="">
                    
    <a class="" href="../QC/">Quality control</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Outline and general concepts for the RNAseq analysis</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../cDNAs/">RNAseq librairies - cDNA synthesis</a>
                </li>
                <li class="">
                    
    <a class="" href="../sequencing_strategies/">RNAseq librairies - Inserts and sequencing strategies</a>
                </li>
                <li class="">
                    
    <a class="" href="../readcounts/">The key idea in Reference-base Expression analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../read_filtering/">Reflecting on quality control & “filtering” in RNAseq analysis</a>
                </li>
                <li class="">
                    
    <a class="" href="../transcript_quant/">Transcript Quantification</a>
                </li>
                <li class="">
                    
    <a class="" href="../RNAseq_DE/">Statistical Analysis of Differential expression</a>
                </li>
                <li class="">
                    
    <a class="" href="../outline_conclusion/">Experimental procedures affect downstream analyses</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Quality filtering</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../Cutadapt/">Optional filtering of reads on sequence quality</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Mapping</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../mapping/">Mapping</a>
                </li>
                <li class="">
                    
    <a class="" href="../star/">RNA STAR</a>
                </li>
                <li class="">
                    
    <a class="" href="../hisat2/">HiSAT2</a>
                </li>
                <li class="">
                    
    <a class="" href="../bam/">BAM inspection</a>
                </li>
                <li class="">
                    
    <a class="" href="../visu_map/">Visualisation of read mapping</a>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Analysis of the differential gene expression</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../intro_counting/">Introduction to read counting</a>
                </li>
                <li class="">
                    
    <a class="" href="../strandness/">Estimation of the strandness</a>
                </li>
                <li class=" current">
                    
    <a class="current" href="./">Count the number of reads per annotated gene</a>
    <ul class="subnav">
            
    <li class="toctree-l3"><a href="#featurecounts">featureCounts</a></li>
    

    <li class="toctree-l3"><a href="#multiqc">MultiQC</a></li>
    

    </ul>
                </li>
                <li class="">
                    
    <a class="" href="../DE_intro/">Identification of the differentially expressed features - DESeq2</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEDESeq2/">DESeq2 use</a>
                </li>
                <li class="">
                    
    <a class="" href="../volcano/">Volcano Plot</a>
                </li>
                <li class="">
                    
    <a class="" href="../DEseq2visu/">Visualization of the differentially expressed genes</a>
                </li>
                <li class="">
                    
    <span class="caption-text">Analysis of functional enrichment among the differentially expressed genes</span>
    <ul class="subnav">
                <li class="toctree-l3">
                    
    <a class="" href="../GO-intro/">Introduction</a>
                </li>
                <li class="toctree-l3">
                    
    <a class="" href="../GO-tool/">GO</a>
                </li>
    </ul>
                </li>
    </ul>
	    </li>
          
            <li class="toctree-l1">
		
    <span class="caption-text">Galaxy Workflows</span>
    <ul class="subnav">
                <li class="">
                    
    <a class="" href="../workflow_intro/">Introduction</a>
                </li>
                <li class="">
                    
    <a class="" href="../worflow_use.md">Example</a>
                </li>
    </ul>
	    </li>
          
        </ul>
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="..">Galaxy Spring day 2019</a>
      </nav>

      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="..">Docs</a> &raquo;</li>
    
      
        
          <li>Analysis of the differential gene expression &raquo;</li>
        
      
    
    <li>Count the number of reads per annotated gene</li>
    <li class="wy-breadcrumbs-aside">
      
    </li>
  </ul>
  <hr/>
</div>
          <div role="main">
            <div class="section">
              
                <p><img alt="" src="../images/galaxylogo.png" /></p>
<h1 id="featurecounts"><code>featureCounts</code></h1>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ol>
<li>In your history <code>HISAT2</code> or <code>STAR</code></li>
<li>Select the <code>featureCounts</code> tool with the following parameters to count your reads:<ol>
<li><code>Alignment file</code>: select multiple datasets button and shift-click the 7 bam files you have generated</li>
<li><code>Specify strand information</code>: <strong>Unstranded</strong></li>
<li><code>Gene annotation file</code> : <strong>in your history</strong><ul>
<li><code>Gene annotation file</code>: <strong>Drosophila_melanogaster.BDGP6.95.gtf</strong></li>
</ul>
</li>
<li><code>FASTA/Q file</code>: <strong>Gene-ID "\t" read-count (MultiQC/DESeq2/edgeR/limma-voom compatible)</strong></li>
<li><code>Create gene-length file</code>: Yes</li>
<li>In <code>Options for paired-end reads</code>:<ul>
<li><code>Count fragments instead of reads</code>: <strong>Enabled; fragments (or templates) will be counted instead of reads</strong></li>
</ul>
</li>
<li>In <code>Advanced options</code>:<ul>
<li><code>GFF feature type filter</code>: <strong>exon</strong></li>
<li><code>GFF gene identifier</code>: gene_id</li>
<li><code>Allow read to contribute to multiple features</code>: No</li>
<li><code>Count multi-mapping reads/fragments</code>: Disabled; multi-mapping reads are excluded (default)</li>
<li><code>Minimum mapping quality per read</code>: 10</li>
</ul>
</li>
<li>Leave other settings as defaults</li>
</ol>
</li>
<li><code>Execute</code></li>
</ol>
<hr />
<p><img alt="" src="../images/rename_datasets.png" /></p>
<p>You need now to rename you datasets to facilitate your downstream analysis.</p>
<p>Be quiet and focus ! No hurry, this is an <em>important task</em> in the analysis.</p>
<ol>
<li>
<p>Search and select datasets with <code>featurecounts</code> (as we did before for renaming datasets)</p>
</li>
<li>
<p>Click on the info icon <img alt="" src="../images/info.png" /> of <code>featureCounts on xxx: Counts</code></p>
</li>
<li>
<p>Copy the name of the dataset <code>Alignment file</code> in the tool parameters table (for instance, <code>GSM461176_untreat_single.bam</code>)</p>
</li>
<li>Now click on the pencil icon of the <em><em>same</em></em> dataset</li>
<li>Paste your text in the <code>Name</code> field of the dataset</li>
<li>Edit your text by replacing <code>bam</code> by <code>Counts</code>
(e.g. GSM461176_untreat_single.Counts)</li>
<li>repeat <em>ad lib</em> for all counts files generated by featureCounts</li>
</ol>
<h1 id="multiqc"><code>MultiQC</code></h1>
<p>We have now generated (1) Bam alignments and (2) Counts files with feature counts, and
we have carefully and courageously edited the names of generated datasets. We are going
to be rewarded for this effort in the next steps !</p>
<hr />
<p><img alt="" src="../images/tool_small.png" /></p>
<ol>
<li>In your history <code>HISAT2</code> or <code>STAR</code></li>
<li>Select the <code>MultiQC</code> tool with the following parameters:<ol>
<li><code>1: Results</code>/ <code>Which tool was used generate logs?</code>: <strong>STAR</strong> <code>or</code> <strong>HISAT2</strong> (depending on your analysis track)</li>
<li><code>STAR or HISAT output</code>: shift-click select all files with the extension <strong>.log</strong></li>
<li>Click on the <code>+ Insert Result</code> button</li>
<li><code>2: Results</code>/ <code>Which tool was used generate logs?</code>: <strong>featureCounts</strong></li>
<li><code>Output of FeatureCounts</code>: shift-click select all files with the extension <strong>: Summary</strong></li>
</ol>
</li>
<li><code>Execute</code></li>
</ol>
<hr />
<p><img alt="" src="../images/oeil.png" /> Examine the results and <img alt="" src="../images/lamp.png" /></p>
              
            </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../DE_intro/" class="btn btn-neutral float-right" title="Identification of the differentially expressed features - DESeq2">Next <span class="icon icon-circle-arrow-right"></span></a>
      
      
        <a href="../strandness/" class="btn btn-neutral" title="Estimation of the strandness"><span class="icon icon-circle-arrow-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
    
  </div>

  Built with <a href="http://www.mkdocs.org">MkDocs</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
      
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" style="cursor: pointer">
    <span class="rst-current-version" data-toggle="rst-current-version">
      
      
        <span><a href="../strandness/" style="color: #fcfcfc;">&laquo; Previous</a></span>
      
      
        <span style="margin-left: 15px"><a href="../DE_intro/" style="color: #fcfcfc">Next &raquo;</a></span>
      
    </span>
</div>
    <script>var base_url = '..';</script>
    <script src="../js/theme.js" defer></script>
      <script src="../search/main.js" defer></script>

</body>
</html>
